plugin/storage/grpc/config/config.go

```diff
@@ -50,60 +76,53 @@ type ClientPluginServices struct {
 // PluginBuilder is used to create storage plugins. Implemented by Configuration.
 // TODO this interface should be removed and the building capability moved to Factory.
 type PluginBuilder interface {
-	Build(logger *zap.Logger, tracerProvider trace.TracerProvider) (*ClientPluginServices, error)
-	Close() error
+	Build(logger *zap.Logger, tracerProvider trace.TracerProvider) (*ClientPluginServices, *grpc.ClientConn, error)
 }
 
 // Build instantiates a PluginServices
-func (c *Configuration) Build(logger *zap.Logger, tracerProvider trace.TracerProvider) (*ClientPluginServices, error) {
-	return c.buildRemote(logger, tracerProvider, grpc.NewClient)
+func (c *Configuration) Build(logger *zap.Logger, tracerProvider trace.TracerProvider) (*ClientPluginServices, *grpc.ClientConn, error) {
+	v2Cfg := c.translateToConfigV2()
+	s, remoteConn, err := newRemoteStorage(v2Cfg, tracerProvider)
+	if err != nil {
+		return nil, nil, err
+	}
+	return s, remoteConn, nil
 }
 
-type newClientFn func(target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error)
+func (c *ConfigV2) Build(logger *zap.Logger, tracerProvider trace.TracerProvider) (*ClientPluginServices, *grpc.ClientConn, error) {
+	s, remoteConn, err := newRemoteStorage(c, tracerProvider)
+	if err != nil {
+		return nil, nil, err
+	}
+	return s, remoteConn, nil
+}
 
-func (c *Configuration) buildRemote(logger *zap.Logger, tracerProvider trace.TracerProvider, newClient newClientFn) (*ClientPluginServices, error) {
+func newRemoteStorage(c *ConfigV2, tracerProvider trace.TracerProvider) (*ClientPluginServices, *grpc.ClientConn, error) {
```

Comment:
Do yo think it's actually important to the caller of this function to get back `*grpc.ClientConn`? I understand that you are returning it in order to ensure it's closed in the end, but from the caller perspective what it's trying to do is instantiate `*ClientPluginServices`. The connection is just an implementation detail of `*ClientPluginServices`, the caller does not need to know about it. Instead we can capture the connection inside `*ClientPluginServices` and have `*ClientPluginServices` implement a Close method.